app/controlplane/pkg/biz/organization_integration_test.go

```diff
@@ -268,3 +272,88 @@ func (s *OrgIntegrationTestSuite) SetupTest() {
 	assert.NoError(err)
 	assert.Len(contracts, 1)
 }
+
+type AuthOnboardingTestSuite struct {
+	testhelpers.UseCasesEachTestSuite
+	usr, usr1 *biz.User
+	org       *biz.Organization
+	m         *biz.Membership
+}
+
+func (s *AuthOnboardingTestSuite) SetupTest() {
+	t := s.T()
+	assert := assert.New(t)
+	ctx := context.Background()
+
+	s.TestingUseCases = testhelpers.NewTestingUseCases(t, testhelpers.WithOnboardingConfiguration([]*conf.OnboardingSpec{
+		{
+			Name: "testing-org",
+			Role: v1.MembershipRole_MEMBERSHIP_ROLE_ORG_VIEWER,
+		},
+	}))
+
+	s.setupUsersAndOrganization(ctx, assert)
+	s.setupMembership(ctx, assert)
+}
+
+func (s *AuthOnboardingTestSuite) setupUsersAndOrganization(ctx context.Context, assert *assert.Assertions) {
+	var err error
+	s.org, err = s.Organization.Create(ctx, "onboarded-org")
+	assert.NoError(err)
+
+	s.usr, err = s.User.CreateByEmail(ctx, "foo@bar")
+	assert.NoError(err)
+
+	s.usr1, err = s.User.CreateByEmail(ctx, "bar@foo")
+	assert.NoError(err)
+}
+
+func (s *AuthOnboardingTestSuite) setupMembership(ctx context.Context, assert *assert.Assertions) {
+	var err error
+	s.m, err = s.Membership.Create(ctx, s.org.ID, s.usr1.ID, biz.WithMembershipRole(authz.RoleViewer))
+	assert.NoError(err)
+}
+
+func (s *AuthOnboardingTestSuite) TestAutoOnboardOrganizations() {
```

Comment:
I'd also test

- The user find or create method calling this new method
- calling this method twice
- calling this without configuration